articles/api/authentication/api-authz/_auth-code-pkce.md

```diff
@@ -0,0 +1,126 @@
+| `response_type` <br/><span class="label label-danger">Required</span> | Indicates to Auth0 which OAuth 2.0 Flow you want to perform. Use `code` for Authorization Code Grant (PKCE) Flow. |
+| `client_id` <br/><span class="label label-danger">Required</span> | Your application's Client ID. |
+| `state` <br/><span class="label label-primary">Recommended</span> | An opaque value the clients adds to the initial request that Auth0 includes when redirecting the back to the client. This value must be used by the client to prevent CSRF attacks. |
+| `redirect_uri` | The URL to which Auth0 will redirect the browser after authorization has been granted by the user. |
+| `code_challenge_method` <br/><span class="label label-danger">Required</span> | Method used to generate the challenge. The PKCE spec defines two methods, `S256` and `plain`, however, Auth0 supports only `S256` since the latter is discouraged. |
+| `code_challenge` <br/><span class="label label-danger">Required</span> | Generated challenge from the `code_verifier`. |
+| `connection` | The name of the connection configured to your application. |
+| `prompt` | To initiate a [silent authentication](/api-auth/tutorials/silent-authentication) request, use `prompt=none` (see Remarks for more info). |
+| `organization` | ID of the [organization](/organizations) to use when authenticating a user. When not provided, if your application is configured to **Display Organization Prompt**, the user will be able to enter the organization name when authenticating. |
+| `invitation` | Ticket ID of the organization invitation. When [inviting a member to an Organization](/organizations/invite-members), your application should handle invitation acceptance by forwarding the invitation and organization key-value pairs when the user accepts the invitation. |
+
+## Get Token
+```http
+POST https://${account.namespace}/oauth/token
+Content-Type: application/x-www-form-urlencoded
+
+grant_type=authorization_code&client_id=${account.clientId}&code_verifier=CODE_VERIFIER&code=AUTHORIZATION_CODE&redirect_uri=${account.callback}
+```
+
+```shell
+curl --request POST \
+  --url 'https://${account.namespace}/oauth/token' \
+  --header 'content-type: application/x-www-form-urlencoded' \
+  --data 'grant_type=authorization_code&client_id=${account.clientId}&code_verifier=CODE_VERIFIER&code=AUTHORIZATION_CODE&redirect_uri=${account.callback}'
+```
+
+```javascript
+var request = require("request");
```

Comment:
This is a bit nitpicky, I'd love to have `const` instead of `var` in the code examples. I think that `var` could transmit the image of being out of fashion, or not up-to-date with the current best practices, which may put off some people.

`var` is perfectly functional in any case, so this is totally optional, but using `const` would transmit a more "modern" feel.

Reply:
@edgarchirivella-okta , these code samples weren't updated this round. Would I just replace `var` with `const`?

Reply:
No problem! I understand that you just moved the code examples between files.

I don't think it's required now, this is something that could be included in the future project you have next quarter, if you have the time to include it 🙂

In these small examples, it would just be replacing `var` with `const`. There are some gotchas because they don't behave in the exact same way, but for these examples, replacing `var` with `const` would do. This is a very good explanation of the differences: https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/

In general, people try to use `const` first, and `let` only if the variable needs to be reassigned. `var` is only seen in older projects nowadays.